dotcom-rendering/src/useEffectReducer.ts

```diff
@@ -0,0 +1,79 @@
+import type { Dispatch } from 'react';
+import { useEffect, useReducer, useRef } from 'react';
+
+type StateAndEffect<State, Effect> = {
+	state: State;
+	effect?: Effect;
```

Comment:
@arelra suggested that the term `Effect` might be confusing, since this is a description of an effect rather than an effect being run, and we may want to call it something else.

Reply:
effectDescription?

Reply:
Would the argument be that an `Effect` would be a function? I’m not sure `EffectDescription` is any more indicative of what is happening. Its only constrain is to be a valid argument for the effect reducer… so you could still pass an actual function!